src/lib/temple/back/dapp.ts

```diff
@@ -467,37 +471,37 @@ async function requestConfirm({ id, payload, onDecline, handleIntercomRequest }:
 }
 
 async function getNetworkRPC(net: TempleDAppNetwork) {
-  const targetRpc =
-    typeof net === 'string' ? TEZOS_NETWORKS.find(n => n.id === net)!.rpcBaseURL : removeLastSlash(net.rpc);
+  if (net === 'sandbox') {
+    return 'http://localhost:8732';
+  }
 
-  if (typeof net === 'string') {
-    try {
-      const current = await getCurrentTempleNetwork();
-      const [currentChainId, targetChainId] = await Promise.all([
-        loadTezosChainId(current.rpcBaseURL),
-        loadTezosChainId(targetRpc).catch(() => null)
-      ]);
-
-      return targetChainId === null || currentChainId === targetChainId ? current.rpcBaseURL : targetRpc;
-    } catch {
-      return targetRpc;
-    }
-  } else {
-    return targetRpc;
+  if (net === 'mainnet') {
+    const rpcUrl = await getActiveTempleRpcUrlByChainId(TEZOS_MAINNET_CHAIN_ID);
+
+    return rpcUrl!;
   }
+
+  if (typeof net === 'string') throw new Error('Unsupported network');
```

Comment:
I think we should support `ghostnet` here as well. However, it requires modifying `TempleDAppNetwork` type